services/order.py

```diff
@@ -0,0 +1,42 @@
+from datetime import datetime
+from typing import Optional
+
+from django.contrib.auth import get_user_model
+from django.db import transaction
+from django.db.models import QuerySet
+
+from db.models import Order, MovieSession, Ticket
+
+User = get_user_model()
+
+
+def create_order(
+    tickets: list[dict[str, int]],
+    username: str,
+    date: Optional[datetime] = None
+) -> None:
+    user = User.objects.get(username=username)
+
+    with transaction.atomic():
+        order = Order.objects.create(user=user, created_at=date)
+
+        if date:
+            Order.objects.filter(id=order.id).update(created_at=date)
+
+        for ticket in tickets:
+            movie_session = MovieSession.objects.get(
+                id=ticket["movie_session"]
+            )
```

Comment:
Similar to the previous comment, consider using `get_object_or_404` or handling the `DoesNotExist` exception when retrieving the movie session by ID. This will help manage cases where the movie session ID does not exist.